src/parser/demonhunter/havoc/modules/spells/BladeDance.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPECS from 'game/SPECS';
+import Events, { DamageEvent, FightEndEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import SpellLink from 'common/SpellLink';
+
+//For Blade dance and Death Sweep
+class BladeDance extends Analyzer {
+  get suggestionThresholds() {
+    return {
+      actual: this.badCast,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  badCast = 0;
+  hitCount = 0;
+  hitTimeStamp: number = 0;
+  firstHitTimeStamp: number = 0;
+  strikeTime: number = 1000;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.spec === SPECS.HAVOC_DEMON_HUNTER;
+    if (!this.active || this.selectedCombatant.hasTalent(SPELLS.TRAIL_OF_RUIN_TALENT) || this.selectedCombatant.hasTalent(SPELLS.FIRST_BLOOD_TALENT)) {
```

Comment:
This says 
if the yare not demon hunter or if they have talent a or talent b active then die

Reply:
You should set active to true or false if the file is active or not meaning put the `hasTalent` calls before the if and set active to whatever it should be

Reply:
Why would you exit out of this module if you have those talents?

Reply:
Changed `this.active = ...` to check you not having those talents instead.

@Pewtro because these bad casts etc should not be calculated if you have any of those 2 talents. These spells should only be in your rotation if you have one of those talents.